src/finch/autoschedule/optimize.py

```diff
@@ -31,19 +34,18 @@ def _lift_subqueries_expr(node: LogicNode, bindings: dict) -> LogicNode:
             return lhs
         case any if any.is_expr():
             return any.make_term(
-                any.head(),
-                *map(lambda x: _lift_subqueries_expr(x, bindings), any.children()),
+                *tuple(_lift_subqueries_expr(x, bindings) for x in any.children()),
             )
         case _:
             return node
 
 
-def lift_subqueries(node: LogicNode) -> LogicNode:
+def lift_subqueries(node: Term) -> LogicNode:
```

Comment:
Should stay `LogicNode` and in all other places in this file.

Reply:
I tried -- that makes the type checker really unhappy. I can try once more though.

Reply:
Done in https://github.com/finch-tensor/finch-tensor-lite/pull/17/commits/bdda4a3d58883b656ba7c5b11fdd0713922797ec.